tests/testlib.bsh

```diff
@@ -429,6 +438,65 @@ begin_expected_fail_test()
   _expected_failure=1 _begin_common_test ${@+"${@}"}
 }
 
+#**
+# .. function:: begin_fail_zone
+#
+# Start a fail zone
+#
+# In practice, having an expected fail or required fail leads to the possibility of a test failing somewhere you don't expect it to. For this reason, fail zones must be denoted in order for :func:`begin_expected_fail_test` and :func:`begin_required_fail_test` tests to succeed, and the the failures must only occur in a fail zone. If a test a failure happens outside the fail zone, the test will be marked as a failure, with the message ``SHOULD HAVE FAILED ELSEWHERE``.
+#
+# Typically this will be called right before the last last line of a test
```

Comment:
I would not that the this tag does not necessarily need to be paired with the end_fail_zone